### ISSUE TITLE
Rewrite doc section about background tasks

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/aiohttp#1104]
Hmm. After rereading your doc change I see `run_all_long_running_tasks` is not needed and using `gather` is confusing.
Why not just add 

```
app.on_startup.append(listen_to_zeromq)
app.on_startup.append(listen_to_redis)
```

Moreover, listeners are probably tasks not just coroutines.
So you should create a task and leave it running.
Also graceful shutdown is very important.

Most likely working code should look like:

```
import contextlib

async def start_backgrounds(app):
    app['redis_listener'] = app.loop.create_task(listen_to_redis())

async def cleanup_backgrounds(app):
    app['redis_listener'].cancel()
    with contextlib.suppress(asyncio.CancelledError):
        await app['redis_listener']

async def listen_to_redis():
    """Listen to messages from Redis Pub/Sub"""
    pass
```

@f0t0n would you provide a PR for doc fix?

**Comment by f0t0n** [comment]
But if we do in this way, the loop inside the Signal will not continue till it awaits for `listen_to_zeromq` and won't get to `listen_to_redis`, right?

```
app.on_startup.append(listen_to_zeromq)
app.on_startup.append(listen_to_redis)
```

 https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/signals.py#L9

**Comment by asvetlov** [comment]
Yes, that's why I've created new task in `start_backgrounds` from my code snippet.
It's encouraged way.
`gather` will wait for coroutines also BTW.

So from my perspective the doc example will wait forever in application initialization stage, isn't it?

**Comment by asvetlov** [comment]
Related to #1092

**Comment by f0t0n** [comment]
I'll create new PR for docs soon.

**Comment by asvetlov** [comment]
thanks

**Comment by f0t0n** [comment]
Hmm, I've just created an example with the code from my first version and it works well:

https://gist.github.com/f0t0n/f5bd3e41a85e70504ce3d249f5e8d14f

**Comment by f0t0n** [comment]
But your example works much better. Namely it allows to register other quick running tasks after you register the long running.

https://gist.github.com/f0t0n/ca8bdef0ade1ba6358abdf8580ed9594

**Comment by asvetlov** [comment]
Yes, sure.
But please keep doc example as short as possible.
The target is providing a simple code snippet but not describing all allowed opportunities.

**Comment by asvetlov** [comment]
Fixed by #1107
